sky/provision/aws/instance.py

```diff
@@ -429,19 +430,75 @@ def _create_node_tag(target_instance, is_head: bool = True) -> str:
             head_instance_id = _create_node_tag(resumed_instances[0])
 
     if to_start_count > 0:
+        target_reservations = (config.node_config.get(
+            'CapacityReservationSpecification',
+            {}).get('CapacityReservationTarget',
+                    {}).get('CapacityReservationId', []))
+        created_instances = []
+        if target_reservations:
+            node_config = copy.deepcopy(config.node_config)
+            node_config['CapacityReservationSpecification'] = {
+                'CapacityReservationTarget': {}
+            }
+
+            reservations = aws_utils.list_reservations_for_instance_type(
+                node_config['InstanceType'], region=region)
+            # Filter the reservations by the user-specified ones, because
+            # reservations contain 'open' reservations as well, which do not
+            # need to explicitly specify in the config for creating instances.
+            target_reservations_to_count = {}
+            for reservation in reservations:
+                if (reservation.targeted and
+                        reservation.name in target_reservations):
+                    target_reservations_to_count[
+                        reservation.name] = reservation.available_resources
+
+            target_reservations_list = sorted(
+                target_reservations_to_count.items(),
+                key=lambda x: x[1],
+                reverse=True)
+            for reservation, reservation_count in target_reservations_list:
+                if reservation_count <= 0:
+                    break
```

Comment:
```suggestion
                    continue
```
should this be `continue` instead?

Reply:
Since we have sorted the list, any later reservations should have less `reservation_count`, so we can break? Just added a comment

Reply:
Good point! Makes sense